CMakeLists.txt

```diff
@@ -373,7 +373,9 @@ if(WITH_OTLP_GRPC
    OR WITH_OTLP_HTTP
    OR WITH_OTLP_FILE)
   find_package(Protobuf)
-  if(Protobuf_VERSION AND Protobuf_VERSION VERSION_GREATER_EQUAL "3.22.0")
+  if(Protobuf_VERSION 
+     AND Protobuf_VERSION VERSION_GREATER_EQUAL "3.22.0" 
+     AND (NOT DEFINED CMAKE_TOOLCHAIN_FILE))
```

Comment:
@owent mentioned, `WITH_ABSEIL` is needed even `CMAKE_TOOLCHAIN_FILE` is defined (like vcpkg is used).

Does this fix the issue? I think we need more details on it.